pkgs/development/libraries/science/math/cudnn/generic.nix

```diff
@@ -25,6 +28,8 @@ assert (hash != null) || (sha256 != null);
 let
   majorMinorPatch = version: lib.concatStringsSep "." (lib.take 3 (lib.splitVersion version));
   version = majorMinorPatch fullVersion;
+  # Use libcublas if available
+  withoutCudaToolkit = libcublas != null;
```

Comment:
What's the reason to have a version with libcublas and another without? Is it purely for backwards compatibility with derivations still based on cudatoolkit?

Reply:
https://github.com/NixOS/nixpkgs/pull/168748#discussion_r851383468

Reply:
Ah ok, so we only have redist packages for CUDA 11.4-11.6 currently. But according to https://developer.download.nvidia.com/compute/cuda/redist/ there are redist packages for all CUDA 11.x versions. 

@FRidh is there a reason that we don't have redist packages for CUDA 11.0-11.3 as well as 11.4-11.6?

Reply:
Can we move this boolean flag to the top-level? We shouldn't introduce more `? null` bug-hiding traps.

Reply:
> @FRidh is there a reason that we don't have redist packages for CUDA 11.0-11.3 as well as 11.4-11.6?

there are unfortunately only manifest files for 11.4 to 11.6

> Can we move this boolean flag to the top-level? We shouldn't introduce more ? null bug-hiding traps.

@SuperSandro2000 could you stop this nonsense. There is nothing wrong with adding `? null` if you handle it. This boolean isn't an option, and is only there because this function is called for several versions, and for older versions the required attribute *does not exist*.

Reply:
Oh I see. There are manifest files for all of 11.x but they are mostly empty up until 11.4.2

Reply:
> There is nothing wrong with adding ? null if you handle it.

It is playing with fire. It is very easy to get things wrong and don't even notice it.